### ISSUE TITLE
OS: Ubuntu: xclip

### ISSUE THREAD

**Issue by xvno** [xvno/blog#156]
## Refs

xclip is a commandline interface to the X selections (clipboard). This tool helps you to copy the output of any command directly into the clipboard and saves you from manually copying and pasting from the terminal. If you have tried copying from the terminal output, you have already realized how tedious the task is. Imagine, if the output is very long, it will be difficult to copy the output manually. This is where xclip tool can benefit you. You can copy the output of any command using this tool. It also allows you to copy the contents of a file directly into the clipboard as well as print the contents of a selection to the standard out.

Installing xclip
xclip is available as a package for Ubuntu so, it can be installed as below. Open a terminal Ctrl + Alt + T and run:

sudo apt-get install xclip
Using xclip
To copy the output of a command into the clipboard, pipe the command into xclip as below:

Long version
ls -la | xclip -selection clipboard
Short version
ls -la | xclip -sel clip
This puts the output of ls -la command into the clipboard, and you can now paste the output into any other program (eg. a text editor) with Ctrl + V outside terminal and Ctrl + Shift + V inside terminal.

To copy the contents of a file (eg. /etc/apt/sources.list) into the clipboard:

Long version
xclip -selection clipboard -in /etc/apt/sources.list
Short version
xclip -sel clip -i /etc/apt/sources.list
To print the contents of the clipboard:

Long version
xclip -selection clipboard -out
Short version
xclip -sel clip -o
To save the contents of the clipboard to a file (eg. ~/myfile.txt):

Long version
xclip -selection clipboard -out > ~/myfile.txt
Short version
xclip -sel clip -o > ~/myfile.txt

**Comment by xvno** [comment]
## how

xclip is a commandline interface to the X selections (clipboard). This tool helps you to copy the output of any command directly into the clipboard and saves you from manually copying and pasting from the terminal. If you have tried copying from the terminal output, you have already realized how tedious the task is. Imagine, if the output is very long, it will be difficult to copy the output manually. This is where xclip tool can benefit you. You can copy the output of any command using this tool. It also allows you to copy the contents of a file directly into the clipboard as well as print the contents of a selection to the standard out.

Installing xclip
xclip is available as a package for Ubuntu so, it can be installed as below. Open a terminal Ctrl + Alt + T and run:

```sh
sudo apt-get install xclip
Using xclip
To copy the output of a command into the clipboard, pipe the command into xclip as below:

Long version
ls -la | xclip -selection clipboard
Short version
ls -la | xclip -sel clip
This puts the output of ls -la command into the clipboard, and you can now paste the output into any other program (eg. a text editor) with Ctrl + V outside terminal and Ctrl + Shift + V inside terminal.

To copy the contents of a file (eg. /etc/apt/sources.list) into the clipboard:

Long version
xclip -selection clipboard -in /etc/apt/sources.list
Short version
xclip -sel clip -i /etc/apt/sources.list
To print the contents of the clipboard:

Long version
xclip -selection clipboard -out
Short version
xclip -sel clip -o
To save the contents of the clipboard to a file (eg. ~/myfile.txt):

Long version
xclip -selection clipboard -out > ~/myfile.txt
Short version
xclip -sel clip -o > ~/myfile.txt
```

**Comment by xvno** [comment]
# Alias
```sh
alias xc="xclip -sel clip"
alias xcat="xclip -sel clip -i"
alias xv="xclip -sel clip -o"
```

**Comment by xvno** [comment]
## Usage

```sh
# copy file.txt
pbcopy file.txt
# paste
xv

# copy output from a formor command
ls -lh | xc
# paste
xv
```
